gozstd.go

```
@@ -185,10 +189,26 @@ func Decompress(dst, src []byte) ([]byte, error) {
 	return DecompressDict(dst, src, nil)
 }
 
+// DecompressLimited appends decompressed src to dst with given limit and returns the result.
```

Please mention in the function docs what happens when the limit is exceeded. I expect the function must return an error in this case instead of returning the first `limit` bytes of the uncompressed data.